analyzers/src/SonarAnalyzer.ShimLayer.CodeGeneration/OperationLightupGenerator.cs

```diff
@@ -1011,23 +1097,39 @@ public InterfaceData(DocumentData documentData, XElement node, ImmutableArray<(s
             {
                 get
                 {
-                    if (this.documentData.Interfaces.TryGetValue(this.BaseInterfaceName, out var baseInterface))
+                    if (this.BaseInterfaceName is not null
+                        && this.documentData.Interfaces.TryGetValue(this.BaseInterfaceName, out var baseInterface))
                     {
                         return baseInterface;
                     }
 
                     return null;
                 }
             }
+
+            public IEnumerable<InterfaceData> InheritedInterfaces
+            {
+                get
+                {
+                    var inheritedInterfaces = new List<InterfaceData>();
+                    for (var baseDefinition = this.BaseInterface; baseDefinition is not null; baseDefinition = baseDefinition.BaseInterface)
+                    {
+                        inheritedInterfaces.Add(baseDefinition);
+                    }
+
+                    inheritedInterfaces.Reverse();
+                    return inheritedInterfaces;
+                }
+            }
         }
 
         private sealed class PropertyData
         {
             public PropertyData(XElement node)
             {
-                this.Name = node.Attribute("Name").Value;
+                this.Name = node.RequiredAttribute("Name").Value;
                 this.AccessorName = this.Name + "Accessor";
-                this.Type = node.Attribute("Type").Value;
+                this.Type = node.RequiredAttribute("Type").Value;
                 this.TypeNonNullable = Type.TrimEnd('?'); // Sonar - When comparing types as strings, the nullable suffix should be ignored.
                 this.WrappedType = TypeNonNullable + "Wrapper"; // Sonar
 
```

Comment:
I think line 1138: `this.IsOverride = node.Attribute("Override")?.Value == "true";` should be marked with `// Sonar` as well